### ISSUE TITLE
Constituent Smoke Test Using SF Framework

### ISSUE THREAD

**Issue by alexfinnarn** [CUCentralAdvancement/sf-digitaleng-sandbox#6]
In #5, Cypress was used in an attempt to work with LWC and the Shadow DOM. However, those attempts never actually succeeded in completing a really simple, three-command constituent creation test.

https://gkushang.github.io/
https://www.testim.io/blog/puppeteer-selenium-playwright-cypress-how-to-choose/

AC:

- [ ] Test with new constituent smoke test: First name, Last name, Constituent type
- [ ] Choose from available drivers: WebDriver, Playwright
- [ ] Write up ADR if the test is easy to write and reliable

**Comment by alexfinnarn** [comment]
```js
export {};
const { I } = inject();
const instanceUrl = process.env.INSTANCE_URL;
const token = process.env.ACCESS_TOKEN;

Given(/Fred creates a constituent/, () => {
  I.amOnPage(`${instanceUrl}/secur/frontdoor.jsp?sid=${token}`);
  I.wait(5);
  I.amOnPage(`${instanceUrl}/lightning/o/Contact/new`);
  I.wait(5);
  I.see('New Constituent: Constituent');
  I.fillField('input[name="firstName"]', 'Johnny');
  I.fillField('input[name="lastName"]', 'Doe');
  // At some point, people made a default for the constituent type...
  // I.click('span[title="Alumnus/a : AL"]');
  // I.click('button[title="Move selection to Chosen"]');
  I.click('button[name="SaveEdit"]');
  I.wait(10);
  I.see('Johnny Doe');
});
```

I was able to create a step definition that tests creating a new constituent. It uses the same login method as with Cypress, but I was concerned about the MFA announcement and if that impacts logins via that method...I think it is under "API/Integration logins" in https://help.salesforce.com/articleView?id=000352937&type=1&mode=1

Now, I need to test this via GH Actions.

**Comment by alexfinnarn** [comment]
😆 😢 ...yeah, this is way harder to do in GH Actions than Cypress. Locally, I have Chrome available to use but not on Github. There might be a GH Action already there to use, but the docs of the codecept thing talk about Sauce Labs. I've used that method before, but it certainly slows things down...

**Comment by alexfinnarn** [comment]
PR with branch in progress: https://github.com/CUCentralAdvancement/testing-codecept/pull/1

**Comment by alexfinnarn** [comment]
https://github.com/salesforce/codeceptjs-bdd/issues/55

**Comment by alexfinnarn** [comment]
I think I got further with this issue and getting the access token in the step definition; however, the Salesforce sandbox used needs changing to `digitalsbx` and I think I used `uciqa` last time. Now, I can't log into `uciqa` and so I might as well do all this in the `digitalsbx` sandbox.

I started writing a readme to make the steps to add a machine user standardized. Salesforce had articles about this but they unpublished them so I turned to a collection of different articles...that I now don't have in front of me.
